src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/location/LocationTest.java

```diff
@@ -0,0 +1,137 @@
+    private static final String UPLOAD_LOCATIONS_URL = BASE_LOCATIONS_URL + "/upload";
+    private static final String GET_LOCATION_BY_ID_URL = BASE_LOCATIONS_URL + "/" + TEST_LOCATION_ID;
+    private static final String GET_LOCATION_BY_NAME_URL = BASE_LOCATIONS_URL + "/name/" + TEST_LOCATION_NAME
+        + "/language/ENGLISH";
+    private static final String GET_LOCATION_BY_REGION_AND_JURISDICTION = BASE_LOCATIONS_URL + "/filter";
+    private static final String DOWNLOAD_CSV_LOCATIONS_URL = BASE_LOCATIONS_URL + "/download/csv";
+
+    @Test
+    void locationControllerHappyPathTests() throws Exception {
+        Map<String, String> headerMap = new ConcurrentHashMap<>();
+        headerMap.put(AUTHORIZATION, BEARER + accessToken);
+
+        Map<String, String> deleteHeaderMap = new ConcurrentHashMap<>();
+        deleteHeaderMap.put(AUTHORIZATION, BEARER + accessToken);
+        deleteHeaderMap.put("x-provenance-user-id", systemAdminProvenanceId);
+
+        Map<String, String> headerMapUploadLocations = new ConcurrentHashMap<>();
+        headerMapUploadLocations.put(AUTHORIZATION, BEARER + accessToken);
+        headerMapUploadLocations.put("Content-Type", MediaType.MULTIPART_FORM_DATA_VALUE);
+
+        String filePath = this.getClass().getClassLoader().getResource("location/ValidLocations.csv").getPath();
+        File csvFile = new File(filePath);
+
+        final Response responseUploadLocations = doPostRequestMultiPart(
+            UPLOAD_LOCATIONS_URL,
+            headerMapUploadLocations, LOCATION_LIST, csvFile
+        );
+
+        assertThat(responseUploadLocations.getStatusCode()).isEqualTo(OK.value());
+        assertEquals(responseUploadLocations.as(Location[].class).length, 3,
+                     "Failed to create all locations");
+        assertThat(responseUploadLocations.asString().contains(TEST_LOCATION_NAME));
+
+        final Response responseGetLocationById = doGetRequest(
+            GET_LOCATION_BY_ID_URL,
+            headerMap
+        );
+        assertThat(responseGetLocationById.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetLocationById.asString().contains(TEST_LOCATION_NAME));
+
+        final Response responseGetAllLocations = doGetRequest(
+            BASE_LOCATIONS_URL,
+            headerMap
+        );
+        assertThat(responseGetAllLocations.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetAllLocations.asString().contains(TEST_LOCATION_NAME));
+        assertThat(responseGetAllLocations.asString().contains("TestLocationTwo"));
+        assertThat(responseGetAllLocations.asString().contains("TestLocationThree"));
+
+        final Response responseGetLocationByName = doGetRequest(
+            GET_LOCATION_BY_NAME_URL,
+            headerMap
+        );
+        assertThat(responseGetLocationByName.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetLocationByName.asString().contains("2001"));
```

Comment:
Better to cast to Location object to make sure 2001 is the location ID

Reply:
Fixed.